File: AkkaDispatcherExample/src/main/java/org/akka/essentials/dispatcher/MsgEchoActor.java
package org.akka.essentials.dispatcher;

import akka.actor.UntypedActor;

public class MsgEchoActor extends UntypedActor {

	int messageProcessed = 0;
	
	@Override
	public void onReceive(Object msg) throws Exception {
		messageProcessed++;
		
		System.out.println(String.format(
				"Received Message '%s' in Actor %s using Thread %s, total message processed %s", msg,
				getSelf().path().name(), Thread.currentThread().getName(),messageProcessed));
	}

}

File: AkkaDispatcherExample/src/main/scala/org/akka/essentials/scala/dispatcher/MsgEchoActor.scala
package org.akka.essentials.scala.dispatcher
import akka.actor.Actor

class MsgEchoActor extends Actor {
  @volatile var messageProcessed:Int = 0

	def receive: Receive = {
		case message =>
			messageProcessed = messageProcessed + 1
      Thread.sleep(100)
			println(
				"Received Message %s in Actor %s using Thread %s, total message processed %s".format( message,
				self.path.name, Thread.currentThread().getName(), messageProcessed))
	}
}